.circleci/config.yml

```diff
@@ -0,0 +1,205 @@
+            - .circleci/ansible/inventory
+      - run:
+          <<: *destroy_device
+          when: on_fail
+
+  # Install the required software like Packer, Virtualbox, etc on bare-metal Packet.net server
+  provision-metal:
+    <<: *job_defaults
+    docker:
+      - image: st2opsadmin/ovabuild-circle
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/ova
+      # Install packer, ovftool, and virtualbox on the bare metal server.
+      # TODO: Eventually this step will not be required since these packages will be part of the custom image (which hasn't been created yet).
+      # TODO: See "Codify the ova build Infrastructure #8": https://github.com/StackStorm/ova/issues/8
+      - run:
+          name: Provision the bare metal server
+          command: ansible-playbook -v -i inventory provision.yml
+          working_directory: .circleci/ansible/
+      - run:
+          name: Sync git workspace to metal server
+          command: rsync -avz -e 'ssh -o StrictHostKeyChecking=no' . root@metal:/home/ova/
+          environment:
+            HOSTALIASES: .circleci/ansible/.hosts
+      - run:
+          <<: *destroy_device
+          when: on_fail
+
+  # Run an OVA build via Packer on bare-metal server
+  build-test-ova:
+    <<: *job_defaults
+    docker:
+      - image: st2opsadmin/ovabuild-circle
+    environment:
+      HOSTALIASES: .circleci/ansible/.hosts
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/ova
+      - run:
+          name: Build & Test OVA
+          command: fab -f .circleci/bin/build.py -H metal build
+      - run:
+          name: Fetch back the produced OVA image
+          command: rsync -avz -e 'ssh -o StrictHostKeyChecking=no' root@metal:/home/ova/builds .
+      - store_artifacts:
+          path: builds
+          destination: .
+      - persist_to_workspace:
+          root: ~/ova
+          paths:
+            - builds
+      - run:
+          <<: *destroy_device
+          when: on_fail
+
+  # TODO: In a future PR, deploy artifact to S3.
+  deploy-ova:
+    <<: *job_defaults
+    docker:
+      - image: st2opsadmin/ovabuild-circle
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/ova
+
+  # Destroy the created Packet.net bare metal device.
+  destroy-metal:
+    <<: *job_defaults
+    docker:
+      - image: st2opsadmin/ovabuild-circle
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/ova
+      - run:
+          <<: *destroy_device
+          name: Destroy bare-metal device
+
+  # Destroy any packet.net bare metal devices running for more than an hour
+  scrap-metal:
+    <<: *job_defaults
+    docker:
+      - image: st2opsadmin/ovabuild-circle
+    steps:
+      - checkout
+      - run:
+          name: Destroy any metal hosts running longer than an hour
+          command: .circleci/bin/scrap-metal
+
+workflows:
+  version: 2
+  ova-cleanup:
+    triggers:
+      - schedule:
+          cron: "0 * * * *"
```

Comment:
After looking at OVA repo CircleCI tasks, I think ideally if we run cleanup workflow once a day,
otherwise it's spamming unnecessary too much.

I'd like to do the same for OVA repo, so this line reminded me about it.

Reply:
Ideally, we wouldn't even need a job to destroy old machines. Doesn't it cost too much to wait up to 24 hours for an idle host to get destroyed?

Reply:
I think it's OK.
Possibility of rare, several hours leaks is acceptable.
Not acceptable is having abandoned instances for days++/months (this happens, y'know :smiley:  ).

Reply:
Let's do it `12h`, as a somewhat middle ground.

Reply:
Sure. I can get on board with that since abandoned instances are fairly rare.